File: Samples/azure-storage/Azure.Java/models/Endpoints.java
/**
 */

package petstore.models;


/**
 * The URIs that are used to perform a retrieval of a public blob, queue or
 * table object.
 */
public class Endpoints {
    /**
     * Gets the blob endpoint.
     */
    private String blob;

    /**
     * Gets the queue endpoint.
     */
    private String queue;

    /**
     * Gets the table endpoint.
     */
    private String table;

    /**
     * Gets the file endpoint.
     */
    private String file;

    /**
     * Get the blob value.
     *
     * @return the blob value
     */
    public String blob() {
        return this.blob;
    }

    /**
     * Set the blob value.
     *
     * @param blob the blob value to set
     * @return the Endpoints object itself.
     */
    public Endpoints withBlob(String blob) {
        this.blob = blob;
        return this;
    }

    /**
     * Get the queue value.
     *
     * @return the queue value
     */
    public String queue() {
        return this.queue;
    }

    /**
     * Set the queue value.
     *
     * @param queue the queue value to set
     * @return the Endpoints object itself.
     */
    public Endpoints withQueue(String queue) {
        this.queue = queue;
        return this;
    }

    /**
     * Get the table value.
     *
     * @return the table value
     */
    public String table() {
        return this.table;
    }

    /**
     * Set the table value.
     *
     * @param table the table value to set
     * @return the Endpoints object itself.
     */
    public Endpoints withTable(String table) {
        this.table = table;
        return this;
    }

    /**
     * Get the file value.
     *
     * @return the file value
     */
    public String file() {
        return this.file;
    }

    /**
     * Set the file value.
     *
     * @param file the file value to set
     * @return the Endpoints object itself.
     */
    public Endpoints withFile(String file) {
        this.file = file;
        return this;
    }

}

File: Samples/azure-storage/Azure.Ruby/azure_storage/models/endpoints.rb
# encoding: utf-8

module Petstore
  module Models
    #
    # The URIs that are used to perform a retrieval of a public blob, queue or
    # table object.
    #
    class Endpoints

      include MsRestAzure

      # @return [String] Gets the blob endpoint.
      attr_accessor :blob

      # @return [String] Gets the queue endpoint.
      attr_accessor :queue

      # @return [String] Gets the table endpoint.
      attr_accessor :table

      # @return [String] Gets the file endpoint.
      attr_accessor :file

      #
      # Validate the object. Throws ValidationError if validation fails.
      #
      def validate
        # Nothing to validate
      end

      #
      # Serializes given Model object into Ruby Hash.
      # @param object Model object to serialize.
      # @return [Hash] Serialized object in form of Ruby Hash.
      #
      def self.serialize_object(object)
        object.validate
        output_object = {}

        serialized_property = object.blob
        output_object['blob'] = serialized_property unless serialized_property.nil?

        serialized_property = object.queue
        output_object['queue'] = serialized_property unless serialized_property.nil?

        serialized_property = object.table
        output_object['table'] = serialized_property unless serialized_property.nil?

        serialized_property = object.file
        output_object['file'] = serialized_property unless serialized_property.nil?

        output_object
      end

      #
      # Deserializes given Ruby Hash into Model object.
      # @param object [Hash] Ruby Hash object to deserialize.
      # @return [Endpoints] Deserialized object.
      #
      def self.deserialize_object(object)
        return if object.nil?
        output_object = Endpoints.new

        deserialized_property = object['blob']
        output_object.blob = deserialized_property

        deserialized_property = object['queue']
        output_object.queue = deserialized_property

        deserialized_property = object['table']
        output_object.table = deserialized_property

        deserialized_property = object['file']
        output_object.file = deserialized_property

        output_object
      end
    end
  end
end
